### PR TITLE
Split TestBasicClient into multiple classes

### PR DESCRIPTION
**Pull request by seanh**
No functional change to the tests, just splitting one test class into multiple. This is going to make these tests easier to extend in future when more tests need to be added.

### CODE REVIEW

File: tests/unit/lms/services/blackboard_api/_basic_test.py [hunk 1]
```diff
@@ -51,8 +51,8 @@ def test_when_the_response_object_is_None(self):
         assert BlackboardErrorResponseSchema(None).parse() == {}
 
 
-class TestBasicClient:
-    def test_get_token(self, basic_client, oauth_http_service):
+class TestGetToken:
```
Comment (jon-betts) [inline tests/unit/lms/services/blackboard_api/_basic_test.py:54]:
```suggestion
class TestBasicClientGetToken:
```

As there are tests for different classes in here (`TestBlackboardErrorResponseSchema`) this is kind of confusing, as we are swapping between two different styles.

File: tests/unit/lms/services/blackboard_api/_basic_test.py [hunk 2]
```diff
@@ -62,6 +62,8 @@ def test_get_token(self, basic_client, oauth_http_service):
             authorization_code=sentinel.authorization_code,
         )
 
+
+class TestRequest:
```
Comment (jon-betts) [inline tests/unit/lms/services/blackboard_api/_basic_test.py:66]:
```suggestion
class TestBasicClientRequest:
```

**Review by jon-betts — CHANGES_REQUESTED** [review]
I'd recommend name spacing the test classes as this is not the only object under test in this module.

Also I'm going to assume this is to make space for future work or something? If this is all there is to it, I'd suggest not doing it. I don't think it adds much as is.

File: tests/unit/lms/services/blackboard_api/_basic_test.py (hunk 1 above)
Comment (seanh) [reply tests/unit/lms/services/blackboard_api/_basic_test.py:54]: :+1: Done

File: tests/unit/lms/services/blackboard_api/_basic_test.py (hunk 2 above)
Comment (seanh) [reply tests/unit/lms/services/blackboard_api/_basic_test.py:66]: :+1:

**Review by jon-betts — APPROVED** [review]

**Pull request merged**